kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/BrokerAddressFilter.java

```diff
@@ -35,71 +33,48 @@ public class BrokerAddressFilter implements MetadataResponseFilter, FindCoordina
-            }
-        }
-
-        public Class<? extends AddressMapping> addressMapper() {
-            return addressMapperClazz;
-        }
-    }
-
-    private final AddressMapping mapping;
-
-    public BrokerAddressFilter(ProxyConfig all, BrokerAddressConfig config) {
-
-        try {
-            this.mapping = config == null || config.addressMapperClazz == null ? new FixedAddressMapping(all)
-                    : config.addressMapper().getDeclaredConstructor(ProxyConfig.class).newInstance(all);
-        }
-        catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-            throw new RuntimeException(e);
-        }
+    public BrokerAddressFilter(ClusterEndpointProvider endpointProvider) {
+        this.endpointProvider = endpointProvider;
     }
 
     @Override
     public void onMetadataResponse(ResponseHeaderData header, MetadataResponseData data, KrpcFilterContext context) {
         for (MetadataResponseBroker broker : data.brokers()) {
-            apply(context, broker, MetadataResponseBroker::host, MetadataResponseBroker::port, MetadataResponseBroker::setHost, MetadataResponseBroker::setPort);
+            apply(context, broker, MetadataResponseBroker::nodeId, MetadataResponseBroker::host, MetadataResponseBroker::port, MetadataResponseBroker::setHost,
+                    MetadataResponseBroker::setPort);
         }
         context.forwardResponse(data);
     }
 
     @Override
     public void onDescribeClusterResponse(ResponseHeaderData header, DescribeClusterResponseData data, KrpcFilterContext context) {
         for (DescribeClusterBroker broker : data.brokers()) {
-            apply(context, broker, DescribeClusterBroker::host, DescribeClusterBroker::port, DescribeClusterBroker::setHost, DescribeClusterBroker::setPort);
+            apply(context, broker, DescribeClusterBroker::brokerId, DescribeClusterBroker::host, DescribeClusterBroker::port, DescribeClusterBroker::setHost,
+                    DescribeClusterBroker::setPort);
         }
         context.forwardResponse(data);
     }
 
     @Override
     public void onFindCoordinatorResponse(ResponseHeaderData header, FindCoordinatorResponseData data, KrpcFilterContext context) {
         for (Coordinator coordinator : data.coordinators()) {
-            apply(context, coordinator, Coordinator::host, Coordinator::port, Coordinator::setHost, Coordinator::setPort);
+            apply(context, coordinator, Coordinator::nodeId, Coordinator::host, Coordinator::port, Coordinator::setHost, Coordinator::setPort);
         }
         context.forwardResponse(data);
     }
 
-    private <T> void apply(KrpcFilterContext context, T broker, Function<T, String> hostGetter, ToIntFunction<T> portGetter, BiConsumer<T, String> hostSetter,
+    private <T> void apply(KrpcFilterContext context, T broker, Function<T, Integer> nodeIdGetter, Function<T, String> hostGetter, ToIntFunction<T> portGetter,
```

Comment:
I'm not wild about `apply` as a name, I'm leaning more towards `translate` or `rewrite` (which are not a lot better). Maybe `exposeAs` but given all the getter and setter method references that feels a bit odd too.

Reply:
Agreed, the name should be improved.  I'm expecting this class to change soon, so let's leave it until then.